### PR TITLE
feat: Claim drop modals and flow

### LINKED ISSUE #8163: Claim drop modals and flow
image

## Abstract
- the flow that we have now has some bugs and does not work how I expected
- here are 2 modals - reused styles from all the modals we already have 
- important is to solve the bugs that were happening with the current version 
   - the major ones: 
      - not disabling the mint button when you have already minted, then showing the notif that one per address limit
      - mint success notif usually closed all the modals 
      - The redirect was quite random as well
- new colors from neutral color changes used

# Design 
updated in confirmation modals page: https://www.figma.com/file/JXNtryaNYhY0XJNArzvPQx/Handoff?type=design&node-id=1458%3A19091&mode=design&t=yg8JpK4u8LeaLaP8-1

## Loading modal
- this one is new, shares some features with the loading modal 
- copy, context, loading, estimate 
- will probably also design error state, we can probably use a similar state as in the auto teleport - instead of trying again there will be a close. 

## Congrats 
- this one is new, lets test this one out. My only concern is that the preview won't load in 30 seconds when the nft is getting minted, is my concern valid? cc @vikiival 
- if it would work nicely, I will slowly push this style to all the congrats messages

top to bottom: 
### transaction
image
- transaction completed, then link to the block explorer, copy button - copies tx link 
### Preview
- biggest part 
image

### Context text 
image
- for the drops, this will be this
- under it is name of the nft and artist 

### Socials 
image
- icons - when hovering over they change to opposite of background color (in light mode its black) 
- copy link copies link to the nft 

### Buttons
- these are new as well, I believe its good for the flow
- you can quickly list or view the nft 

image

list opens list modal, view nft gallery item

## Drop page changes
- I would like to change the cta button copy a bit: 
- "Claim NFT Now" 
- it's more informative about what would happen

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

drop ahk/drops/chaosflakes

- [x] Closes #8163

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

CleanShot 2023-12-06 at 16 38 47@2x

CleanShot 2023-12-06 at 16 23 16@2x

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 9936f52</samp>

This pull request adds and refactors components and functions to improve the user experience and functionality of the NFT drop and minting process. It introduces new modals, skeleton loaders, countdown timers, and social sharing features. It also updates some translation keys, CSS classes, and types. The main files affected are `Generative.vue`, `GalleryItemShareBtn.vue`, `en.json`, and `waifu.ts`.

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 9936f52</samp>

> _We're minting NFTs with `DropConfirmModal`_
> _We're sharing them on X and Telegram_
> _We're heaving away on the count of three_
> _We're the finest coders on the sea_

### CODE REVIEW

**Comment by hassnian** [comment]
@vikiival 

need nft name on the same response , could you please add that ?

also listing part not sure how to approach it , 
CleanShot 2023-12-06 at 16 44 09@2x

the nft is null for a while , I suppose the transaction is not confirmed or takes some time 

qa @prury 

visual check @exezbcz 

thanks

**Comment by vikiival** [comment]
> need nft name on the same response , could you please add that ?

Added :) 

> also listing part not sure how to approach it ,

Please do not paste screenshot snippets, use either permalinks to existing code or [Creating and highlighting code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks)

> the nft is null for a while

Please share you address, can investigate.

**Comment by yangwao** [comment]
> i was thinking have the button of listing disabled , and enable it when nft is available

I thought that list button should appear when you got confirmed tx?

**Comment by prury** [comment]
- i think view NFT button could also lead to a newtab, wdyt @exezbcz ? this new modal is so cool i don't want to make it go away
- mobile: [image](https://github.com/kodadot/nft-gallery/assets/36627808/5e252085-fff8-4ce3-b5e2-6d17f1fafac4)
- Collection link leads nowhere: [image](https://github.com/kodadot/nft-gallery/assets/36627808/fcbec8c4-5f07-4693-9c49-d1d685cf6a2e)

@kodadot/internal-dev 
is there a way to verify if registered e-mail already minted drop? otherwise i can easily create multiple accounts and register the same e-mail, maybe it could be yet another blocker to scavengers, does it make sense even with the 1.2 DOT ED requirement?

**Comment by vikiival** [comment]
> is there a way to verify if registered e-mail already minted drop?

I can set limitation that email should be unique

File: components/shared/SkeletonLoader.vue [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<template>
+  <div class="h-full skeleton-container">
+    <NeoSkeleton
+      class="skeleton-backdrop"
+      rounded
+      border-radius="20px"
```
Comment (roiLeo) [inline components/shared/SkeletonLoader.vue:6]: 😢

**Review by roiLeo — APPROVED** [review]

File: components/shared/SkeletonLoader.vue (hunk 1 above)
Comment (hassnian) [reply components/shared/SkeletonLoader.vue:6]: ? could you elaborate. thanks

File: components/collection/drop/modal/SuccessfulDrop.vue [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  shareOnX(sharingTxt, nftFullUrl.value)
+}
+
+const handleShareOnTelegram = () => {
+  shareOnTelegram(sharingTxt, nftFullUrl.value)
+}
+</script>
+
+<style scoped>
+.circle {
```
[the remark this answers is not in the document]
Comment (hassnian) [reply components/collection/drop/modal/SuccessfulDrop.vue]: > small prop is not showing excepted result?

nope

>  or use is-size-7

need smaller added .is-size-8

File: components/shared/SkeletonLoader.vue [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      no-margin
+      full-size
+      variant="k-grey-light" />
+
+    <div class="skeleton-content is-flex">
+      <NeoIcon
+        icon="spinner-third"
+        class="spinner has-text-k-grey mr-6"
+        size="large"
+        spin></NeoIcon>
```
[the remark this answers is not in the document]
Comment (hassnian) [reply components/shared/SkeletonLoader.vue]: fixed

**Comment by prury** [comment]
> @prury please also check mobile version as majority (60%) is coming on drops from mobile, NovaWallet mostly and less SubWallet, thanks! 🙏🏾

i did test on firefox using mobile simulation, let me test on nova

**Comment by hassnian** [comment]
> > need nft name on the same response , could you please add that ?
> 
> Added :)

thanks, updated it on my end. 

> 
> > also listing part not sure how to approach it ,
> 
> Please do not paste screenshot snippets, use either permalinks to existing code or [Creating and highlighting code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks)
> 

duly noted 

> > the nft is null for a while
> 
> Please share you address, can investigate.

right after getting the response from do waifu


```
    const { result } = await doWaifu(
      {
        address: accountId.value,
        metadata: hash,
        image: imageHash,
        email,
      },
      props.drop.id,
    )
```

i tried getting the nft 

````

await fetchNft(mintedNft.value?.id)

````

and it would return null , for like 30s to 1 min aprox , after that it would return the mft , that is to be expected , right ?

**Comment by vikiival** [comment]
> and it would return null , for like 30s to 1 min aprox , after that it would return the mft , that is to be expected , right ?

Yes

File: components/shared/SkeletonLoader.vue (hunk 1 above)
Comment (roiLeo) [reply components/shared/SkeletonLoader.vue:6]: I thought I asked on a previous PR to unify border radius on skeleton

**Comment by prury** [comment]
Nova drop minting experience(Android 13): 

https://github.com/kodadot/nft-gallery/assets/36627808/fb292a05-c80f-4939-9391-25ae422aed30

- keyboard in the way of our modal (unrelated)
- sharing on twitter and other buttons prob do not open new tab
- view nft redirects to ahp address, but i guess its intended since chaosflakes is the only one on ahk, just need someone to check

File: components/shared/SkeletonLoader.vue (hunk 1 above)
Comment (roiLeo) [reply components/shared/SkeletonLoader.vue:6]: I got no problem if you do so in a follow up

**Comment by Jarsen136** [comment]
> view nft redirects to ahp address, but i guess its intended since chaosflakes is the only one on ahk, just need someone to check

You did find an important bug. I just tested on two prefix for the same account. I actually mint two nfts at the same address at the same time. 

correct one: https://deploy-preview-8440--koda-canary.netlify.app/ahk/drops/chaosflakes
incorrect prefix: https://deploy-preview-8440--koda-canary.netlify.app/ahp/drops/chaosflakes
image

A simple redirect solution:  Check the current prefix, If it does not match the current drop, we could redirect to the correct one.

Feel free to raise another issue for it or solve it here.


Going further, we should probably adjust the way we currently check addresses, since the same account will have different addresses on different networks. cc @vikiival

**Comment by exezbcz** [comment]
> is there a way to verify if registered e-mail already minted drop? otherwise i can easily create multiple accounts and register the same e-mail, maybe it could be yet another blocker to scavengers, does it make sense even with the 1.2 DOT ED requirement?

plus this 
- https://github.com/kodadot/nft-gallery/issues/8371

**Comment by exezbcz** [comment]
> * I think view NFT button could also lead to a newtab

we can close since you can list the nft there as well

**Comment by hassnian** [comment]
> > i was thinking have the button of listing disabled , and enable it when nft is available
> 
> I thought that list button should appear when you got confirmed tx?

https://github.com/kodadot/nft-gallery/assets/44554284/e559a210-7819-4038-be0e-a2f17033df2b

@prury fixed all mentioned issues, please test again thanks

File: components/collection/drop/Generative.vue [hunk 4]
```diff
@@ -104,15 +105,19 @@ import { NeoButton, NeoIcon } from '@kodadot1/brick'
...
 import { sanitizeIpfsUrl } from '@/utils/ipfs'
 import newsletterApi from '@/utils/newsletter'
+import DropConfirmModal from './modal/DropConfirmModal.vue'
+import ListingCartModal from '@/components/common/listingCart/ListingCartModal.vue'
+import { nftToListingCartItem } from '@/components/common/shoppingCart/utils'
+import { fetchNft } from '@/components/items/ItemsGrid/useNftActions'
 
 const NuxtLink = resolveComponent('NuxtLink')
-const MINTING_SECOND = 120
+const MINTING_SECOND = 59
```
Comment (vikiival) [inline components/collection/drop/Generative.vue:120]: Longer for sue

**Comment by prury** [comment]
> > view nft redirects to ahp address, but i guess its intended since chaosflakes is the only one on ahk, just need someone to check
> 
> You did find an important bug. I just tested on two prefix for the same account. I actually mint two nfts at the same address at the same time.
> 
> correct one: https://deploy-preview-8440--koda-canary.netlify.app/ahk/drops/chaosflakes incorrect prefix: https://deploy-preview-8440--koda-canary.netlify.app/ahp/drops/chaosflakes image
> 
> A simple redirect solution: Check the current prefix, If it does not match the current drop, we could redirect to the correct one.
> 
> Feel free to raise another issue for it or solve it here.
> 
> Going further, we should probably adjust the way we currently check addresses, since the same account will have different addresses on different networks. cc @vikiival

not sure if i understood your message @Jarsen136

**Comment by yangwao** [comment]
> please check my video from **2:50s** onward

but that is with 59 seconds, not 120secs?

**Comment by yangwao** [comment]
> tried again and its still showing 60sec

maybe cache? delete in application all site cache? tried in incognito as well?

**Comment by hassnian** [comment]
> > tried again and its still showing 60sec
> 
> maybe cache? delete in application all site cache? tried in incognito as well?

checking

**Comment by prury** [comment]
tried now on chrome after deleting cache twice, and first attempt there:

image

**Comment by hassnian** [comment]
CleanShot 2023-12-07 at 18 49 49@2x
 seems fixed

**Comment by yangwao** [comment]
if prury will give green, we will do release, even launch is at 3pm we won't make it at start

**Comment by prury** [comment]
> if prury will give green, we will do release, even launch is at 3pm we won't make it at start

checking

**Comment by prury** [comment]
it mints top fast here 😄 

https://github.com/kodadot/nft-gallery/assets/36627808/b5ee2bf3-9c4e-4352-b6f4-e4678fd523c9

i still think we will need to tweak it in the future
if anyone else could test would be nice

**Comment by yangwao** [comment]
> wdyt @yangwao, release?

that 8 second delay can make meh impression tbh 

I'll give 10 minutes if @hassnian will figure out some fix and let's see

meanwhile go claim drop https://kodadot.xyz/ahp/drops/motherboard 

image

**Comment by yangwao** [comment]
thanks!
pay 150 usd

made follow up
- #8456

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $150 @ 6.01 USD/DOT ~ 24.958 $DOT
🧗 [13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh](https://polkadot.subscan.io/account/13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)
🔗 [0xb1a66cad1d554bc49e339fcedcd215f5b76eca01650c4b108e81dc876ff82205](https://polkadot.subscan.io/extrinsic/0xb1a66cad1d554bc49e339fcedcd215f5b76eca01650c4b108e81dc876ff82205)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**